src/Alert/Alert.tsx

```diff
@@ -0,0 +1,194 @@
+        color: theme === "light" ? colors.black.base : colors.white,
+        boxShadow: `0 4px 8px 0 rgba(0, 0, 0, .04)`,
+        borderStyle: "solid",
+        borderRadius: 4,
+        borderWidth: theme === "light" ? 1 : 0,
+        borderColor: colors.silver.dark,
+        padding: 15,
+      }}
+    >
+      <div
+        css={{
+          marginBottom: extended ? 14 : 6,
+          overflow: "hidden",
+          display: "flex",
+        }}
+      >
+        <ClassNames>
+          {({ css, cx }) => {
+            const headingProps = {
+              className: cx(
+                css({
+                  fontWeight: 600,
+                  marginBottom: 0,
+                  marginTop: 0,
+                  width: "100%",
+                  display: "flex",
+                  color: getOffsetInPalette(
+                    2,
+                    theme === "light" ? "darker" : "lighter",
+                    color
+                  ),
+                  ...base.base,
+                })
+              ),
+              children: (
+                <Fragment>
+                  <Icon
+                    css={{ width: 20, height: 20, color, marginRight: 13 }}
+                  />
+                  {heading}
+                </Fragment>
+              ),
+            };
+
+            return React.isValidElement(headingAs)
+              ? React.cloneElement(headingAs, {
+                  ...headingProps,
+                  className: classnames(
+                    headingProps.className,
+                    headingAs.props.className
+                  ),
+                })
+              : React.createElement(headingAs, headingProps);
+          }}
+        </ClassNames>
+        <IconClose
+          onClick={onClose}
+          css={{
+            color: colors.grey.lighter,
+            cursor: "pointer",
+            width: 10,
+            height: 10,
+          }}
+        />
+      </div>
+
+      {extended && (
+        <hr
+          css={{
+            height: 1,
+            borderWidth: 0,
+            backgroundColor:
+              theme === "dark" ? colors.grey.dark : colors.silver.dark,
+            marginTop: 14,
+            marginBottom: 14,
+          }}
+        />
+      )}
+      <div css={{ marginLeft: extended ? 0 : 33 }}>
+        <div
+          css={{
+            fontSize: 13,
+            lineHeight: "1.5",
+            marginBottom: actions ? 13 : 0,
+          }}
+        >
+          {children}
+        </div>
+
+        {actions}
+      </div>
+    </section>
+  );
+};
+
+Alert.propTypes = {
+  extended: PropTypes.bool,
+  onClose: PropTypes.func.isRequired,
+  children: PropTypes.node,
+  heading: PropTypes.node.isRequired,
+  actions: PropTypes.node,
+  color: PropTypes.oneOf(
+    Object.values(colors)
+      .map((color) => Object.values(color))
+      .reduce((a, b) => a.concat(b)) as PaletteColor[]
+  ).isRequired,
+  Icon: PropTypes.func.isRequired,
```

Comment:
I'm asking for this to be changed elsewhere so you don't have to change this here, but this will fail the check if `Icon` is a class component.